### PR TITLE
Get the iface name from the guest os

### PR DESCRIPTION
**Pull request by yalzhang**
Get the iface name from the guest os instead of assume it's interface name.

### CODE REVIEW

**Comment by yalzhang** [comment]
@chloerh  Please help to check, Thank you!

File: libvirt/tests/cfg/virtual_network/passt/passt_connectivity_between_2vms.cfg [hunk 1]
```diff
@@ -36,15 +36,13 @@
             ipv6_addr_c = '2001:db8:ac10:fd01::20'
             ips = {'ips': [{'address': '${ipv4_addr_c}', 'family': 'ipv4', 'prefix': '24'}, {'address': '${ipv6_addr_c}', 'family': 'ipv6'}]}
             iface_c_attrs = {'model': 'virtio', 'acpi': {'index': '1'}, **${ips}, 'backend': ${backend}, 'source': {'dev': '${host_iface}'}, 'type_name': 'user'}
-            vm_c_iface = eno1
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virtual_network/passt/passt_connectivity_between_2vms.cfg:39]: Look like vm_c_iface is referred to by  conn_check_args_1 = ('TCP6', server_default_gw_v6, vm_c_iface, 41335, 41335)

Comment (yalzhang) [reply libvirt/tests/cfg/virtual_network/passt/passt_connectivity_between_2vms.cfg:39]: Yes, and  the "conn_check_args_1" above is referred as "*args" below:
passt.check_protocol_connection(client_session, server_session,*args)
while check_protocol_connection is like this:
check_protocol_connection(src_sess, tar_sess, protocol, addr, src_iface=None, src_port=None, tar_port=None, expected=True, expect_msg=None)

**Review by chloerh — APPROVED** [review]

**Review by Yingshun — APPROVED** [review]

**Review by cliping — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]
lgtm

**Pull request merged**